pvlib/clearsky.py

```diff
@@ -182,14 +182,9 @@ def lookup_linke_turbidity(time, lats_1d, lons_1d, grid=False, filepath=None,
     # daynumber for every 15-th day of the month; including a leading dec and trailing jan
     daynumber = get_daynumber(time)
```

Comment:
You might even revert my `daynumber` changes and just do `daynumber = [-15, 15, 46, 74, 105, 135, 166, 196, 227, 258, 288, 319, 349, 379]`. The `linke_turbidity_data` is a fixed dataset that is time/year independent. So does it make sense to adjust the `daynumber` in leap years?

Reply:
It probably does not matter much, but also the original upstream version without Floris' changes last year has a leap year treatment:
https://github.com/pvlib/pvlib-python/blob/13b0134442690af8e2f20055a2145cc34acd8331/pvlib/clearsky.py#L269
So I will keep it

Reply:
The idea is that the h5 data file contains data for the "middle of the month" and which day that is depends on whether we have a leap year.

Reply:
I agree that it doesn't matter. The whole thing is an approximation and even the middle of month is just an assumption (https://github.com/pvlib/pvlib-python/blob/13b0134442690af8e2f20055a2145cc34acd8331/pvlib/clearsky.py#L248).
My idea was that we could get rid of an additional function. But with the function it might be a little bit cleaner.